### ISSUE TITLE
Feature request: self-contained environment

### ISSUE THREAD

**Issue by nedsociety** [sandboxie-plus/Sandboxie#1736]
### Is your feature request related to a problem or use case?

Isolating and analyzing multiple environments (like, per app basis) requires you to create a sandbox, run it with settings, and then inspect the Sandbox directory for each app. This pattern is not self-contained, prone to error and awkward to automate. The only automation support from Sandboxie right now is the process execution by `ForceFolder`, but it would be great to have the rest as well.

### Describe the solution you'd like

The proposal is to create a self-contained sandbox based on a local directory configuration. If a directory contains a ".sandboxie" subdirectory, it indicates that the programs under that directory is supposed to be sandboxed.

Add an option under [Global Settings/Program Control] section: "Enable self-contained sandboxes". If this option is checked, invoke a global behavior:

* When a process spawns, if its executable path has an ancestor that contains a subdirectory called `.sandboxie`, its process tree is automatically isolated under a temporarily created sandbox.

  * For safety, the following ancestors are excluded from qualifying the existence of `.sandboxie`:
    * `%PROGRAMFILES%`, `%PROGRAMFILES(X86)%`, `%SystemDrive%`, `%SystemRoot%` and `%TEMP%`

* The `.sandboxie` directory must have a file "Sandboxie.ini" containing a single section called `[SelfContainedSandbox]` with the sandbox settings. These options are applied to the temporary sandbox.

* The temp sandbox works like an ordinary sandbox with following quirks:
  * The name is uniquely determined by its path.
  * It works as if the global option `FileRootPath` is overridden by `<path to .sandboxie directory>/Sandbox/%USER%`.
  * In addition to the root `Sandbox` directory, the sandboxed processes are denied from seeing/accessing the corresponding `.sandboxie` directory.
  * It's temporary. It is registered under the GUI managers as long as it's running, and removed from the list when it terminates.

**Comment by DocMAX** [comment]
So this should make an app in a folder "portable", right? Then i second this.

**Comment by mysteriously** [comment]
TLDR: Treat sandbox folder as forced folder

**Comment by DavidXanatos** [comment]
I'm not sure about the eficiency of always checking the entire ancestral tree of a executable path, but i see how this could be usefull.
Unfortunately it requires additional changes to how sandboxie manages internally configurations. etc... so its rather a large change.
But that change would be useful for some other features I'm planning so this feature may come in the upcoming months.

**Comment by jterrain** [comment]
Thanks all - what is the implementation status on this concept?
